File: ai_server/notes/dlls_demo/interview/repository/interview_repository.py
from abc import ABC, abstractmethod
from typing import List

class InterviewRepository(ABC):
    @abstractmethod
    def generateQuestions(
        self, interview_id: int, topic: str, experience_level: str, user_token: str
    ) -> str:
        pass

    @abstractmethod
    def generateFollowupQuestion(
        self, interview_id: int, question_id: int, answer_text: str, user_token: str
    ) -> str:
        pass

File: m/notes/first/lib/interview/infrastructures/repository/interview_repository.dart
import '../../domain/entity/interview.dart';
import '../../domain/usecases/create/response/interview_create_response.dart';
import '../../domain/usecases/create/response/interview_followup_response.dart';
import '../../domain/usecases/list/response/interview_list_response.dart';

abstract class InterviewRepository {
  Future<InterviewListResponse> listInterview(int page, int perPage, String userToken);
  Future<InterviewCreateResponse> createInterview(String userToken, Interview interview);
  Future<InterviewFollowupResponse?> createInterviewQuestionAnswer({
    required String userToken,
    required int interviewId,
    required int questionId,
    required String answerText,
  });
}
